src/NHapi.SourceGeneration/NormativeDatabase.cs

```diff
@@ -1,185 +1,191 @@
-				while (rs.Read())
-				{
-					Object tabNum = rs.GetValue(1 - 1);
-					Object val = rs.GetValue(3 - 1);
-					Object desc = rs.GetValue(4 - 1);
-					Console.Out.WriteLine("Table: " + tabNum + " Value: " + val + " Description: " + desc);
-				}
-			}
-			catch (DbException e)
-			{
-				log.Error("test msg!!", e);
-			}
-			catch (Exception e)
-			{
-				log.Error("test msg!!", e);
-			}
-		}
-
-		static NormativeDatabase()
-		{
-			log = HapiLogFactory.GetHapiLog(typeof(NormativeDatabase));
-		}
-	}
+    using System;
+    using System.Data;
+    using System.Data.Common;
+    using System.Data.Odbc;
+    using System.Diagnostics;
+
+    using NHapi.Base.Log;
+
+    /// <summary>
+    /// Point of access to a copy of the HL7 normative database.
+    /// </summary>
+    /// <example>
+    /// A typical way of obtaining and using a database connection would be ...
+    /// <para>
+    /// <code>
+    /// Connection c = NormativeDatabase.getInstance().getConnection();<br />
+    /// // ... use the database ... <br />
+    /// NormativeDatabase.returnConnection(c);
+    /// </code>
+    /// </para>
+    /// <para>
+    /// Since the database may be installed differently on different systems, certain system
+    /// properties must be set with the required connection information, as follows: <br />
+    /// <code>ca.on.uhn.hl7.database.url</code> - the URL of the JDBC connection.<br />
+    /// <code>ca.on.uhn.hl7.database.user</code> - the user ID needed to connect (if required).<br />
+    /// <code>ca.on.uhn.hl7.database.passsword</code> - the password associated with the above user (if required).<br />
+    /// </para>
+    /// <para>
+    /// The required JDBC driver must also be loaded (this can be done by ensuring that the
+    /// required driver appears in the class path and appending the class name to the
+    /// "jdbc.drivers" system property.
+    /// </para>
+    /// </example>
+    /// <author>Bryan Tripp (bryan_tripp@sourceforge.net).</author>
+    public class NormativeDatabase
+    {
+        private static readonly IHapiLog Log;
+
+        private static NormativeDatabase db = null;
+        private string connectionString;
+
+        /// <summary> Returns the singleton instance of NormativeDatabase.  </summary>
+        private OdbcConnection odbcConnection;
+
+        static NormativeDatabase()
+        {
+            Log = HapiLogFactory.GetHapiLog(typeof(NormativeDatabase));
+        }
+
+        /// <summary> Private constructor ... checks system properties for connection
+        /// information.
+        /// </summary>
+        private NormativeDatabase()
+        {
+            connectionString = ConfigurationSettings.ConnectionString;
+            odbcConnection = new OdbcConnection(connectionString);
+            odbcConnection.Open();
+        }
+
+        public static NormativeDatabase Instance
+        {
+            get
+            {
+                lock (typeof(NormativeDatabase))
```

Comment:
We can deffo use Lazy T here since the source generator doesn't need to support net35, if it doesnt currently we can probably remove net35 from the target framework.

Reply:
dont do this as part of this ticket as ill take a look at source generation at somepoint

Reply:
ok - have left a `TODO` comment for now